calyx-py/test/correctness/queues/strict_and_rr_queues/gen_strict_or_rr.py

```diff
@@ -158,34 +167,14 @@ def insert_queue(
                 ],
             ),
             len_decr,
-            restore_hot if not round_robin else ast.Empty            
-        ],
-    )
-
-    peek_logic = cb.if_with(
-        len_eq_0,
-        raise_err,  # The queue is empty: underflow.
-        [  # The queue is not empty. Proceed.
-            raise_err,  # We raise err so we enter the loop body at least once.
-            copy_hot,  # We remember `hot` so we can restore it later.
-            [
-                cb.while_with(
-                    err_is_high,
-                    [  # We have entered the loop body because `err` is high.
-                        # Either we are here for the first time,
-                        # or we are here because the previous iteration raised an error
-                        # and incremented `hot` for us.
-                        # We'll try to peek from `fifo_cells[hot]`.
-                        # We'll pass it a lowered `err`.
-                        lower_err,
-                        invoke_subqueues_hot_guard,
-                        incr_hot_wraparound,  # Increment hot: this will be used
-                        # only if the current subqueue raised an error,
-                        # and another iteration is needed.
-                    ],
-                ),
-            ],
-            restore_hot,  # Peeking must not affect `hot`, so we restore it.
+            (
+                pifo.reg_store(hot, og_hot.out)
+                if not round_robin
+                else ast.Empty
+                # If we are not generating a round-robin PIFO,
+                # we are generating a strict PIFO.
+                # We need to restore `hot` to its original value.
+            ),
```

Comment:
This is how I avoided the unused group error in case of RR. I add the group into the component `pifo` only if the Python-level boolean `round_robin` is false!